src/lib.rs

```diff
@@ -716,6 +692,9 @@ pub trait Log: Sync + Send {
     /// Implementations of `log` should perform all necessary filtering
     /// internally.
     fn log(&self, record: &Record);
+
+    /// Flushes any buffered records.
+    fn flush(&self) {}
```

Comment:
The [issue]( https://github.com/rust-lang-nursery/log/issues/117) mentioned `flush` ought to have a default implementation, but I'm happy to remove it here. This obviously has ramifications for `env_logger`; would its implementation be expected to use the same reference counting pattern previously used here when coordinating on `shutdown`?